src/common/user-store/preferences-helpers.ts

```diff
@@ -222,6 +237,31 @@ const syncKubeconfigEntries: PreferenceDescription<KubeconfigSyncEntry[], Map<st
   },
 };
 
+const editorConfiguration: PreferenceDescription<EditorConfiguration, EditorConfiguration> = {
+  fromStore(val) {
+    const result: EditorConfiguration = {};
+
+    result.lineNumbers = val?.lineNumbers ?? defaultEditorConfig.lineNumbers;
+    result.tabSize = val?.tabSize ?? defaultEditorConfig.tabSize;
+    result.miniMap = val?.miniMap ?? defaultEditorConfig.miniMap;
+
+    return result;
+  },
+  toStore(val) {
+    const result: EditorConfiguration = {};
+
+    if (val.lineNumbers != defaultEditorConfig.lineNumbers)
+      result.lineNumbers = val.lineNumbers;
+    if (val.tabSize != defaultEditorConfig.tabSize)
+      result.tabSize = val.tabSize;
+    if (val.miniMap?.enabled != defaultEditorConfig.miniMap?.enabled)
+      result.miniMap = val.miniMap;
+
+    return result;
+  },
```

Comment:
Why we compare settings with default ones before saving them to store? I guess we need to skip this comparisons.

Reply:
I requested this because the goal was to be able to change the defaults in the future.

Reply:
Probably it's not a good idea to create this logic for some future possible scenarios. I suggest using migrations for such cases.